README.md

```diff
@@ -105,25 +113,60 @@ Sometimes you need a service to be available alongside your Jupyter server. In t
 
 A service is described by 
 
-* **template** - a name of a template imported in OpenShift which describes how to deploy the service
-* **parameters** - an object containing key-value pairs to be processed into the template
+* **resources** - these are the names of configMaps which contain various templates. For each resource there can be a different configMap with a different path. Or we can use a single configMap containing all templates and these are simply taken by using the correct path.
+* **configuration** - an object containing key-value pairs to be processed into the template
+* **labels** - object containing labels that will be proccessed into the template
 * **return** - an object containing key-value pairs where value is a [JSON Path](https://github.com/kennknowles/python-jsonpath-rw) walkable in the uploaded ConfigMap/CustomResource and key is a name of environment variable under which the value will be available in Jupyterhub Singleuser server
 
 A `USERNAME` parameter is automatically added based on user's name to separate services by user.
 
+As long as the resource is correctly defined in the template and there is a custom resource definition on the cluster, the template can be used to create any custom resource, or a configMap.
+
+In this example, we call this template configMap containing `sparkClusterTemplate` that is entered into the `path` variable.
```

Comment:
This sentence is weird. How about

The following example shows a ConfigMap containing `sparkClusterTemplate` object which can be used as a `path` reference in the `service` definition.